src/MOL_symbolic_utils.jl

```diff
@@ -244,3 +255,15 @@ function ex2term(term, v)
     name = Symbol("⟦" * string(term) * "⟧")
     return setname(similarterm(exdv, rename(operation(exdv), name), arguments(exdv)), name)
 end
+
+safe_unwrap(x) = x isa Num ? unwrap(x) : x
+
+function recursive_unwrap(ex)
```

Comment:
Ideally, we won't need this. We should just call unwrap when constructing the tree.

Reply:
I ran in to a case with dot where it won't work without Num wrapping, and you already need to recursively unwrap the subtree. But yes once the missing `dot(a, b::AbstractArray{<:Symbolic})` method exists I'd like to get rid of this.